src/schema.ts

```diff
@@ -76,22 +86,33 @@ export const resolvers = {
       return findUser;
     },
 
-    users: async (_, list: { numUsers: number }, context: Context): Promise<UserList> => {
+    users: async (_, list: { data: UserQueryInput }, context: Context): Promise<UserList> => {
       const secret = process.env.JWT_SECRET ?? 'secret';
       jwt.verify(context.token, secret);
 
       const DEFAULT_NUMBER_OF_USERS = 10;
-      const takeUsers = list.numUsers ?? DEFAULT_NUMBER_OF_USERS;
+      const takeUsers = list.data.numUsers ?? DEFAULT_NUMBER_OF_USERS;
+      const skipUsers = list.data.skip ?? 0;
 
       const userRepository = getRepository(User);
-      const usersList = await userRepository.find({ order: { name: 'ASC' }, take: takeUsers });
+      const response = await userRepository.findAndCount({ order: { name: 'ASC' }, skip: skipUsers, take: takeUsers });
+      const usersList = response[0];
+      const totalUsers = response[1];
+
+      const users = usersList.length;
+
+      let hasPreviousPage = true;
+      if (skipUsers === 0 || usersList.length === 0) {
+        hasPreviousPage = false;
+      }
```

Comment:
Aqui vc pode usar direto a expressão do if pra definir o seu `hasPreviousPage`, porque ela é uma expressão booleana, né?

```suggestion
      const hasPreviousPage = skipUsers === 0 || usersList.length === 0;
```

Reply:
Entendi, vou refatorar essa parte, mas para a lógica estar certa, precisa de uma negação, pois `hasPreviousPage` é falso se está no início da lista ou se a lista é vazia

Reply:
Ah, boa, fiz errado. Vou editar com a lógica correta.